src/icalendar/prop.py

```diff
@@ -238,13 +238,22 @@ def from_ical(cls, ical):
 class vDDDLists(object):
     """A list of vDDDTypes values.
     """
-    def __init__(self, dt_list):
+    def __init__(self, dt_list, type_class=None):
+        if type_class is None:
+            type_class = vDDDTypes
         if not hasattr(dt_list, '__iter__'):
             dt_list = [dt_list]
         vDDD = []
         tzid = None
+        if dt_list:
+            # we have some values, all should have the same type
+            ltype = type(dt_list[0])
         for dt in dt_list:
-            dt = vDDDTypes(dt)
+            # raise ValueError if type of the input values differs
+            if not isinstance(dt, ltype):
```

Comment:
Hm, you're right, it would seem that we need to be even more strict here. Thanks for pointing that out with the `date`/`datetime` example, I did not realize.